metrics/prometheus-metrics/src/main/scala/sttp/tapir/metrics/prometheus/PrometheusMetrics.scala

```diff
@@ -0,0 +1,136 @@
+      output.close()
+      output.toString
+    })
+
+  def withDefaultMetrics[F[_]](implicit monad: MonadError[F]): PrometheusMetrics[F] =
+    PrometheusMetrics[F](
+      defaultRegistry,
+      Set(
+        requestsTotal(defaultRegistry),
+        requestsActive(defaultRegistry),
+        requestsFailures(defaultRegistry),
+        responsesTotal(defaultRegistry),
+        responsesDuration(defaultRegistry)
+      )
+    )
+
+  def requestsTotal[F[_]](registry: CollectorRegistry)(implicit monad: MonadError[F]): Metric[F, Counter] =
+    Metric[F, Counter](
+      Counter
+        .build()
+        .namespace("tapir")
+        .name("requests_total")
+        .help("Total HTTP requests")
+        .labelNames("path", "method")
+        .create()
+        .register(registry)
+    ).onRequest { (req, counter) => monad.unit(counter.labels(path(req), method(req)).inc()) }
+
+  def requestsActive[F[_]](registry: CollectorRegistry)(implicit monad: MonadError[F]): Metric[F, Gauge] =
+    Metric[F, Gauge](
+      Gauge
+        .build()
+        .namespace("tapir")
+        .name("requests_active")
+        .help("Active HTTP requests")
+        .labelNames("path", "method")
+        .create()
+        .register(registry)
+    ).onRequest { (req, gauge) => monad.unit(gauge.labels(path(req), method(req)).inc()) }
+      .onResponse { (req, _, gauge) => monad.unit(gauge.labels(path(req), method(req)).dec()) }
+
+  def requestsFailures[F[_]](registry: CollectorRegistry)(implicit monad: MonadError[F]): Metric[F, Counter] =
+    Metric[F, Counter](
+      Counter
+        .build()
+        .namespace("tapir")
+        .name("requests_failure")
+        .help("Unserved HTTP requests")
+        .labelNames("path", "method")
+        .register(registry)
+    ).onDecodeFailure { (req, counter) => monad.unit(counter.labels(path(req), method(req)).inc()) }
+
+  def responsesTotal[F[_]](registry: CollectorRegistry)(implicit monad: MonadError[F]): Metric[F, Counter] =
+    Metric[F, Counter](
+      Counter
+        .build()
+        .namespace("tapir")
+        .name("responses_total")
+        .help("HTTP responses")
+        .labelNames("path", "method", "status")
+        .register(registry)
+    ).onResponse { (req, res, counter) => monad.unit(counter.labels(path(req), method(req), status(res)).inc()) }
+
+  def responsesDuration[F[_]](registry: CollectorRegistry)(implicit monad: MonadError[F]): Metric[F, Histogram] =
+    Metric[F, Histogram](
+      Histogram
+        .build()
+        .namespace("tapir")
+        .name("responses_duration")
+        .help("HTTP responses duration")
+        .labelNames("path", "method", "status")
+        .register(registry)
+    ).onResponse { (req, res, histogram) =>
+      monad.unit(histogram.labels(path(req), method(req), status(res)).observe((Deadline.now - req.timestamp).toMillis.toDouble / 1000.0))
+    }
+
+  private def path(request: ServerRequest): String = request.pathSegments.mkString("/")
```

Comment:
We shouldn't use the full request's path, but rather the endpoint's path template. That way, any dynamic url segments are replaced with path captures